libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -491,17 +491,27 @@ bool AP_Arming::gps_checks(bool report)
             }
         }
 
-        //GPS OK?
-        if (!AP::ahrs().home_is_set() ||
-            gps.status() < AP_GPS::GPS_OK_FIX_3D) {
-            check_failed(ARMING_CHECK_GPS, report, "Bad GPS Position");
-            return false;
-        }
+        for (uint8_t i = 0; i < gps.num_sensors(); i++) {
+            if (gps.get_type(i) == AP_GPS::GPS_Type::GPS_TYPE_NONE) {
+                if (gps.primary_sensor() == i) {
+                    check_failed(ARMING_CHECK_GPS, report, "GPS primary not configured");
```

Comment:
Doesn't this block you from disabling GPS if you don't need it? This looks like it will have the indoor positioning people yelling at us...

Reply:
Yes it does, but you already have to turn off the gps check or you fail at hdop, see https://github.com/ArduPilot/ardupilot/issues/14661

IMHO we should not be comming into the gps check at all in that case

Reply:
That sorta breaks the template here: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Arming/AP_Arming.cpp#L1215

If we want to not run GPS checks then we should be bailing in the start of GPS checks. (How to reliably ID that is a different discussion)